### PR TITLE
feat: FT-39 manage themes

### PR DESCRIPTION
**Pull request by LiamMartens**

### CODE REVIEW

File: src/app/components/Button.tsx [hunk 1]
```diff
@@ -1,84 +1,111 @@
...
-  switch (size) {
-    case 'large':
-      sizeClass = 'button-large';
-      break;
-    default:
-      break;
-  }
+  }, [id, onClick]);
+
+  const variantClass = React.useMemo(() => (
```
Comment (six7) [inline src/app/components/Button.tsx:55]: Could we move away from variant classes and instead use Stitches' variants? That should be way easier to work with.

File: src/app/components/EmptyState/EmptyState.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import Text from '../Text';
+import { StitchesCSS } from '@/types';
+
+type Props = React.HTMLAttributes<HTMLDivElement> & {
+  css?: StitchesCSS
+  title: string
+  subtitle?: string
+};
+
+export const EmptyState: React.FC<Props> = ({ title, subtitle, ...props }) => (
```
Comment (six7) [inline src/app/components/EmptyState/EmptyState.tsx:12]: There's the `Blankslate` component already, could we either consolidate this with the other one or just use the other one?

**Review by six7 — COMMENTED** [review]
This is great already! I added two minor comments

File: src/app/components/Button.tsx (hunk 1 above)
Comment (LiamMartens) [reply src/app/components/Button.tsx:55]: @six7 yea the button situation is something I want to ask you - I don't even think the variant classes work anymore so I wanted to check

File: src/app/components/EmptyState/EmptyState.tsx (hunk 2 above)
Comment (LiamMartens) [reply src/app/components/EmptyState/EmptyState.tsx:12]: @six7 of course, ill have a look - I didnt realize it existed

File: src/app/components/Button.tsx (hunk 1 above)
Comment (six7) [reply src/app/components/Button.tsx:55]: i'm currently integrating a loading state into the <Button> - i can refactor it separately, no need to do that in this PR. i'll take your changes from this PR

**Comment by six7** [comment]
One thing that would be helping users:
- When hitting `Create theme` could we take the current token sets as a basis? So whatever is set there, is set in the Create theme dialog? That way a user would not have to recreate everything from scratch when they create a new theme, but could use what they have currently as a basis. E.g. going from dark to light mode would mean only swapping one token set.

**Comment by LiamMartens** [comment]
@six7 I updated the remote storage providers to support additional files for themes etc.. also re-worked it to be more generic and interchangeable for the future. -- I want to get this functionally checked and merged into next without the figma style ID logic because it's getting pretty large

**Review by six7 — APPROVED** [review]

**Pull request merged**